slack-rtm.c

```diff
@@ -187,8 +187,15 @@ static gboolean rtm_connect_cb(SlackAccount *sa, gpointer data, json_value *json
 	slack_blist_init(sa);
 
 	slack_login_step(sa);
+
+	size_t cookie_size = strlen("d=") + strlen(sa->d_cookie) + 1;
+	char *cookie = malloc(cookie_size);
+	snprintf(cookie, cookie_size, "d=%s", sa->d_cookie);
```

Comment:
FWIW, since we're using glib already, it'd be simpler to just do let it figure out the buffer size automatically. IOW, those three lines can just simplify to

```
   gchar *cookie = g_strdup_printf("d=%s", sa->dcookie ? sa->d_cookie : "");
```

Reply:
Even if sa->d_cookie is null?

Reply:
I guess you are missing the `_` separator in your suggestion while accessing d_cookie the first time.

Reply:
> FWIW, since we're using glib already, it'd be simpler to just do let it figure out the buffer size automatically. IOW, those three lines can just simplify to
> 
> ```
>    gchar *cookie = g_strdup_printf("d=%s", sa->d_cookie ? sa->d_cookie : "");
> ```
> 

Or 
``` 
gchar *cookie = g_strconcat("d=", sa->d_cookie, NULL);
``` 
:D 

(do cookie values need url escaping?)

Reply:
Thanks, actually we should not send the cookie header at all if it's missing, so I took this into account as well. See latest push.